### PR TITLE
Log out after system is registered

### PR DESCRIPTION
**Pull request by pdostal**
- Related ticket: [poo#170356](https://progress.opensuse.org/issues/170356)
- Verification run: [SLES](https://pdostal-server.suse.cz/tests/7776), [SAP](https://pdostal-server.suse.cz/tests/7784)

### CODE REVIEW

**Review by rbmarliere — APPROVED** [review]

**Comment by m-dati** [comment]
I think some VR are needed here

**Review by asmorodskyi — APPROVED** [review]
LGTM

**Review by mpagot — APPROVED** [review]
LGTM

File: lib/publiccloud/utils.pm [hunk 1]
```diff
@@ -116,6 +116,7 @@ sub registercloudguest {
     # Check what version of registercloudguest binary we use
     $instance->ssh_script_run(cmd => "rpm -qa cloud-regionsrv-client");
     $instance->ssh_script_retry(cmd => "sudo $suseconnect -r $regcode", timeout => 420, retry => 3, delay => 120);
+    assert_script_run('ssh -O exit ' . $instance->username . '@' . $instance->public_ip);
```
Comment (grisu48) [inline lib/publiccloud/utils.pm:119]: I think we need to do this on every console that is opened.

Comment (pdostal) [reply lib/publiccloud/utils.pm:119]: I think your suggestion applies only to `TUNNELED=1` tests. But even in those we don't (re)register the system after `publiccloud/ssh_interactive_start` test module.

The `publiccloud::utils::registercloudguest` subroutine is currently used in the following test modules:
- `publiccloud/registration`
- `publiccloud/migration`
- `publiccloud/slem_basic`
- `publiccloud/run_ltp`
- `publiccloud/check_registercloudguest`
- `publiccloud/storage_perf`
- `sles4sap/publiccloud/qesap_terraform`
- `sles4sap/cloud_netconfig/deploy`

I think think that we are safe. But please feel free to suggest more VRs.

**Comment by pdostal** [comment]
@m-dati SAP VR done.

**Review by m-dati — APPROVED** [review]

File: lib/publiccloud/utils.pm (hunk 1 above)
Comment (grisu48) [reply lib/publiccloud/utils.pm:119]: No you're right. Let's proceed.

**Review by grisu48 — APPROVED** [review]

**Comment by alvarocarvajald** [comment]
This seems to have broken saptune tests, across all CSPs:

- https://openqa.suse.de/tests/16112457#step/registration/23
- https://openqa.suse.de/tests/16112516#step/registration/23
- https://openqa.suse.de/tests/16112462#step/registration/23

HanaSR jobs are OK, but saptune jobs (which are more similar to QE-C jobs) are the ones that are failing.

**Pull request merged**